histogram.js

```diff
@@ -72,27 +75,39 @@ class HistogramElement extends HTMLElement {
     this._histo = core({
       title: this.label,
     });
-    
     this.shadowRoot.appendChild(this._histo.el);
-
     const broker = getDataBroker(this);
+
+    const loader = this.shadowRoot.querySelector('.samplingLoader');
+
+    broker.onEvent('data-request-start', () => {
+      loader.style.display = 'block';
+      this._histo.update([]);
+    });
+
+    broker.onEvent('data-streaming-start', () => {
+        loader.style.display = 'none';
+    });
+
     if (broker) {
       let data = [];
       this._histo.update(data);
-
       broker.onEvent(this.brokerKey, (data) => {
-
-        var d = Object.keys(data).filter(function (i) {
-          return data[i] != "0"
-        }).map(function (k) {
-          return [+k, +data[k]]
-        });
-
-        if (this.ignoreOutliers) {
-          d = iobioviz.layout.outlier()(d);
+        if (loader.style.display === 'block') {
```

Comment:
Why is this check necessary? The event shouldn't be triggered unless there's data coming in, and if there's data coming in, the loader shouldn't be showing.

Reply:
Ah that makes sense. In that case I would switch to using a local variable as mentioned above, but one with a good name that represents the states we're switching between. Maybe `dataReady`?